src/librustc/middle/region.rs

```diff
@@ -467,9 +467,10 @@ impl<'tcx> Visitor<'tcx> for ExprLocatorVisitor {
     }
 
     fn visit_pat(&mut self, pat: &'tcx Pat) {
+        intravisit::walk_pat(self, pat);
```

Comment:
This makes the visitation match the visitation in `RegionResolutionVisitor`. I don't think this matters for patterns. It might mess up generators inside patterns somehow though.